### PR TITLE
Add a FolioClient service

### PR DESCRIPTION
**Pull request by corylown**
Set up for #3135

### CODE REVIEW

File: app/services/folio_client.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    authenticated_request(path, method: :post, **kwargs)
+  end
+
+  def post_json(path, **kwargs)
+    parse(post(path, **kwargs))
+  end
+
+  # FOLIO Edge API - Real Time Availability Check
+  # https://s3.amazonaws.com/foliodocs/api/edge-rtac/p/edge-rtac.html
+  def rtac(instance_ids:, full_periodicals: true)
```
Comment (jcoyne) [inline app/services/folio_client.rb:53]: Can we call this `real_time_availability`?

File: app/services/folio_client.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    parse(get(path, **kwargs))
+  end
+
+  def post(path, **kwargs)
+    authenticated_request(path, method: :post, **kwargs)
+  end
+
+  def post_json(path, **kwargs)
+    parse(post(path, **kwargs))
+  end
```
Comment (jcoyne) [inline app/services/folio_client.rb:49]: Can these be private?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/services/folio_client.rb (hunk 1 above)
Comment (corylown) [reply app/services/folio_client.rb]: yes, sounds good

File: app/services/folio_client.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'http'
+
+class FolioClient
+  DEFAULT_HEADERS = {
+    accept: 'application/json, text/plain',
```
Comment (jcoyne) [inline app/services/folio_client.rb:7]: If we're going to parse JSON from the response, then `text/plain` isn't really acceptable to us.

**Review by jcoyne — APPROVED** [review]

File: spec/services/folio_client_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe FolioClient do
+  subject(:client) { described_class.new(url:) }
+
+  let(:url) { 'https://example.com' }
```
Comment (jcoyne) [inline spec/services/folio_client_spec.rb:8]: Should a different host be considered, because example.com is in the allowed list?

**Pull request merged**